torch/csrc/utils/tensor_new.cpp

```diff
@@ -264,6 +283,10 @@ Tensor internal_new_from_data(
     return var.to(device, inferred_scalar_type, /*non_blocking=*/false, /*copy=*/copy_variables);
   }
 
+  auto sizes = compute_sizes(data);
```

Comment:
Add a comment here about why sizes and inferred_scalar_type is computed here.

Ideally they would be computed closer to where they're used, since the code immediately below these definitions shadows infered_scalar_type and doesn't use size.

Reply:
I have moved the computation of ```infered_scalar_type``` into the blocks where it is being used.